Add Debezium Channel Adapter

tzolov opened the pull request:
Resolves #3779

tzolov committed:
Add Debezium Channel Adapter

  Resolves #3779

artembilan commented on build.gradle:
```diff
@@ -464,6 +465,26 @@ project('spring-integration-test-support') {
+    description = 'Spring Integration Debezium Support'
+    dependencies {
+        api project(':spring-integration-core')
+        api("io.debezium:debezium-api:$debeziumVersion") {
```
Cannot we reply on a `debezium-embedded` dep?
Why do we need this extra `debezium-api` one?

We also need to think about excluding those EE deps.
See GH action failure for PR where it shows duplicated classpath entries.

You can run that locally like `./gradlew :spring-integation-debezium checkClasspathForConflicts` commenting out its `onlyIf { isCI }`

artembilan commented on build.gradle:
```diff
@@ -464,6 +465,26 @@ project('spring-integration-test-support') {
+        }
+
+        testImplementation("io.debezium:debezium-connector-mysql:$debeziumVersion"){
+            exclude group: 'org.slf4j', module: 'slf4j-reload4j'
```
Do we really see the problem with this `slf4j-reload4j`?
Why cannot we try to rely on a default state of dependency until we face a problem like in case of those duplicates mentioned above

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,256 @@
+
+/**
+ *
+ * @author Christian Tzolov
```
`@since 6.2` and some Javadocs.

tzolov replied:
sure

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,256 @@
+ */
+public class DebeziumMessageProducer extends MessageProducerSupport implements BeanClassLoaderAware {
+
+	static final Log logger = LogFactory.getLog(DebeziumMessageProducer.class);
```
There is already a `logger` property in the super `IntegrationObjectSupport`

tzolov replied:
+

artembilan replied:
I think I said that before, but just to be sure that we understood each other.
I prefer to have just a commit with the change requested instead of commit and confirmation comments like this.
This way there is no need for you to remember to answer to me for this my comment and there is no an overhead in my mail box and I don't waste time reading this type of comments.
With the next commit I will be sure that whatever was reviewing is incorporated over there or will be done later on.

If you don't agree with my review, then you are more than welcome to discuss and complain.

Of course this is optional: just wanted to let you know how we can optimize our collaboration 😄

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,256 @@
+			// Note: Event for none flattened responses, when the debezium.properties.tombstones.on.delete=true
+			// (default), tombstones are generate by Debezium and handled by the code below.
+			if (payload == null) {
+				payload = DebeziumMessageProducer.this.kafkaNull;
```
I think we need to get rid of Kafka mentioning in this module.
Make this type of payload as `Optional.empty()` instead.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DefaultDebeziumHeaderMapper.java:
```diff
@@ -0,0 +1,58 @@
+ *
+ * @author Christian Tzolov
+ */
+public class DefaultDebeziumHeaderMapper<T> implements HeaderMapper<List<Header<T>>> {
```
Why do we need this generic argument?
Doesn't look like we really care about header value when we map - `messageHeaders.put(key, value)`.
Do we need to worry about them? Convert them into String, for example?

Also: this class has to go into top-level package: this is not just an `inbound` feature

artembilan commented on spring-integration-debezium/src/test/resources/logback-test.xml:
```diff
@@ -0,0 +1,30 @@
+  -->
+
+<configuration>
+	<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
```
We use `log4j2-test.xml` in this project.
Also: no Copyright in the XML files, please.

artembilan reviewed (changes requested):
I left some review, but there is no need in a prompt fixing: we are going to look into this again when we switch to `6.2.0` version next week after releasing `6.1.0`.

Thank you for great PR!

tzolov committed:
initial debezium doc

tzolov commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,256 @@
+				this.latch.countDown();
+			}
+		});
+		this.latch = new CountDownLatch(1);
```
+

tzolov committed:
address some reviews

tzolov committed:
resolve some classpath conflicts

tzolov committed:
hacking failed test

tzolov committed:
fixing tests and dependecies

tzolov committed:
address review comments

tzolov committed:
improve test coverage

tzolov committed:
fix test checkstyle

tzolov committed:
remove kafak references. hit support for batch

tzolov committed:
improve java doc

tzolov committed:
Initial batch support

 Convert the list of Change events into list of Messages.
 Use the same rules for buidling messages as the non-batch mode.

tzolov committed:
Refine batch implementation and tests

tzolov committed:
harden the testcontainers start/stop lifecycle

tzolov committed:
simplify batch mode

tzolov committed:
adjust test log config

tzolov committed:
clean gradle config

artembilan commented on spring-integration-debezium/src/test/resources/log4j2-test.xml:
```diff
@@ -0,0 +1,21 @@
+			<AppenderRef ref="Console" />
+		</Root>
+	</Loggers>
+</Configuration>
```
Please, ensure that every file comes with a new line in the end.
Would be great if your IDE could do that on save.

tzolov committed:
Add  HeaderMapper filter configuration. Fix JavaDocs

tzolov committed:
Use CustomizableThreadFactory for Exec Service. IT header tests

artembilan commented on build.gradle:
```diff
@@ -575,6 +576,29 @@ project('spring-integration-core') {
+            exclude group: 'org.glassfish.jersey.containers', module: 'jersey-container-servlet'
+            exclude group: 'org.glassfish.jersey.inject', module: 'jersey-hk2'
+            exclude group: 'jakarta.xml.bind', module: 'jakarta.xml.bind-api'
+            exclude group: 'jakarta.activation', module: 'jakarta.activation-api'
```
Any updates here, please?

tzolov replied:
All dependencies listed above are violating the `checkClasspathForConflicts`. 
Have not observed any side effects excluding them. 
The integration tests work as expected, so I also assume that they are used only if the Debezium engine is run in certain (RH) web applications.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumHeaders.java:
```diff
@@ -0,0 +1,39 @@
+ * limitations under the License.
+ */
+
+package org.springframework.integration.debezium.inbound;
```
Headers abstraction and respective mapper must go to the `support` package.
Even if we don't have any outbound channel adapter it is better to be consistent with other modules structure.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+ */
+public class DebeziumMessageProducer extends MessageProducerSupport {
+
+	private DebeziumEngine.Builder<ChangeEvent<byte[], byte[]>> debeziumEngineBuilder;
```
`final`

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+
+	private boolean enableBatch = false;
+
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
```
Some Javadoc, please.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
+		Assert.notNull(debeziumBuilder, "Failed to resolve Debezium Engine Builder. " +
+				"Debezium Engine Builder must either be set explicitly via constructor argument.");
```
This message does not make sense here.
Just "cannot be null" is enough.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+	 * "https://debezium.io/documentation/reference/2.2/transformations/event-flattening.html#extract-new-record-state-add-headers-prefix">add.headers.prefix</a>
+	 *
+	 */
+	public void setAllowedHeaderNames(String... headerNames) {
```
I don't think this is a channel adapter responsibility.
Please, consider to move this into `DefaultDebeziumHeaderMapper` impl.
Well, I see you have it there already anyway.
But then just a setter for a `HeaderMapper` must be provided here on adapter.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+
+	/**
+	 * Defines if a single or a batch of messages are send downstream.
+	 *
```
Method Javadocs are imperative.
No blank lines in method Javadocs, please: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,263 @@
+	 * Set the {@link ExecutorService}, where is not provided then a default of single thread Executor will be used.
+	 * @param executorService the executor service.
+	 */
+	public void setExecutorService(ExecutorService executorService) {
```
Any updates here?

tzolov replied:
If not mistaken the only purpose of the Spring TaskExecutor is to provide backward (per JDk 1.6) compatibility with java.util.concurrent.Executor? If so it is an anahornism that should be deprecated from new code.

Regarding the Executor vs ExecutorService, later offers submit() and Features support, offereing higher degree of control over the running Featrues. What benefits would it bring to degradate it to a plain Executor instead?

artembilan replied:
That's true. I also like an `ExecutorService` better, but there is just no an `ExecutorService` auto-configuration in Spring Boot. See a `TaskExecutionAutoConfiguration`.
It produces for us a `ThreadPoolTaskExecutor` which is an impl of `SchedulingTaskExecutor` which, in turn is an `AsyncTaskExecutor`, but not an `ExecutorService`.
So, we are stuck with Spring API until it is resolved the way you think.

My point is that with the current logic we just just don't need a `Future` hook, but just that `DebeziumEngine.close()`.

However the more I think about this and the more we move to virtual threads, it is probably better to not pursue an external managed executor, but rather let end-user to supply a custom `ThreadFactory`, where the virtual variant can be chosen.
See the `PartitionedChannel` impl.
The point is that we always need just one thread for that  `DebeziumEngine` to work and such a thread is always taken when we run.
So, with an external executor we would need to add yet a note that thread is going to be taken for lifecycle of this channel adapter.

Therefor I'm taking my review back and suggest to inject a `ThreadFactory` instead.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+	 * {@link DebeziumEngine}.
+	 */
+	public void setContentType(String contentType) {
+		this.contentType = contentType;
```
`Assert.hasText()`?

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+			this.executorService.shutdown();
+		}
+		try {
+			this.executorService.awaitTermination(10, TimeUnit.SECONDS);
```
This is not OK for externally provided executor.
We really cannot control how that one is shouted down.
We may wait on an internal one, but why

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+
+	private <T> Message<?> toMessage(ChangeEvent<T, T> changeEvent) {
+
+		if (logger.isDebugEnabled()) {
```
No need in `is..Enabled` since `LogAccessor` can use lambdas for lazy strings creation.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,289 @@
+		public void accept(ChangeEvent<T, T> changeEvent) {
+			sendMessage(toMessage(changeEvent));
+		}
+	}
```
New line before the last brace in the class.
Any class, even if it anonymous.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DefaultDebeziumHeaderMapper.java:
```diff
@@ -0,0 +1,106 @@
+
+	private static final String[] DEBEZIUM_HEADER_NAMES = {
+			DebeziumHeaders.DESTINATION,
+			DebeziumHeaders.KEY
```
But as we know from channel adapter impl that these props are not header in a `ChangeEvent`.
So, it useless to look for them here the logic of this class.

artembilan commented on spring-integration-debezium/src/test/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducerTests.java:
```diff
@@ -0,0 +1,115 @@
+import static org.mockito.Mockito.mock;
+
+/**
+ * @author Christian Tzolov
```
I also prefer to point the version test class was introduce, so `@since 6.2`

artembilan commented on spring-integration-debezium/src/test/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducerTests.java:
```diff
@@ -0,0 +1,115 @@
+
+	DebeziumEngine.Builder<ChangeEvent<byte[], byte[]>> debeziumBuilderMock;
+	DebeziumEngine<ChangeEvent<byte[], byte[]>> debeziumEngineMock;
+	DebeziumMessageProducer debeziumMessageProducer;
```
Every member in the class must be surrounded with blank line, even the last method.
This really makes it much easier to read.

I wonder if we need to start looking into Java Formatter support as many other portfolio does: https://github.com/spring-io/spring-javaformat.

This way after `format` all the code becomes consistent and questions like this won't arise 😄

artembilan commented on spring-integration-debezium/src/test/java/org/springframework/integration/debezium/inbound/DefaultDebeziumHeaderMapperTests.java:
```diff
@@ -0,0 +1,119 @@
+		}
+
+		@SuppressWarnings("unchecked")
+		public TestHeader(org.apache.kafka.connect.header.Header header) {
```
Do we really need this Kafka-based ctor in this mock class?

artembilan commented on spring-integration-debezium/src/test/resources/log4j2-test.xml:
```diff
@@ -0,0 +1,21 @@
+		<Logger name="com.github.dockerjava" level="WARN"/>
+		<Logger name="org.springframework.integration.debezium" level="info"/>
+
+		<Logger name="org.junit.vintage.engine.discovery" level="error" />
```
I think we need to revise these categories, since I believe some of them just copy-n-paste from somewhere else.

artembilan commented on src/reference/asciidoc/debezium.adoc:
```diff
@@ -0,0 +1,105 @@
+[[debezium-inbound]]
+=== Inbound Debezium Channel Adapter
+
+The Debezium adapter expects a pre-configured `DebeziumEngine.Builder<ChangeEvent<byte[], byte[]>>` bean instance. The following code snippets demonstrate various configuration for this channel adapter:
```
Any updates here?

artembilan commented on src/reference/asciidoc/index.adoc:
```diff
@@ -38,6 +38,7 @@ Welcome to the Spring Integration reference documentation!
 <<./amqp.adoc#amqp,AMQP Support>> :: AMQP channels, adapters and gateways
 <<./camel.adoc#camel,Apache Camel Support>> :: Apache Camel channel adapters and gateways
 <<./cassandra.adoc#cassandra,Apache Cassandra Support>> :: Apache Cassandra channel adapters
+<<./debezium.adoc#debezium,Debezium CDC Support>> :: Debezium inbound channel adapters
```
Just `Debezium `channel adapters` - even if we have only one for now.

artembilan reviewed (changes requested):
We also need a `whats-new.adoc` bullet for a new chapter.
There is also a `endpoint-summary.adoc` with the table of all the channel adapters the project.
So, a new line for Debezium must be added there as well.

There is also a failed test: 
```

DebeziumMessageProducerTests > testDebeziumMessageProducerLifecycle() FAILED
    org.mockito.exceptions.verification.WantedButNotInvoked at DebeziumMessageProducerTests.java:84
```

tzolov committed:
review stuff

tzolov committed:
review fixes and doc improvements

tzolov committed:
more debeizum documentation

tzolov committed:
Replace ExecutorService by Executor

tzolov committed:
remove redundant log configs

tzolov committed:
Remove external Executor support in favour of configurable ThreadFactory

artembilan commented on build.gradle:
```diff
@@ -575,6 +576,24 @@ project('spring-integration-core') {
+        testImplementation "io.debezium:debezium-connector-mysql:$debeziumVersion"
+
+        testImplementation project(':spring-integration-stream')
+        testImplementation 'org.springframework:spring-web'
```
Any justification for these two deps for our tests?

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,283 @@
+	}
+
+	/**
+	 * Defines if a single or a batch of messages are send downstream.
```
Still missed an imperative style change.
Method Javadocs must start like `Define` - kinda command to do.

tzolov replied:
Still miss to understand what you want so feel free to write it the way if find it fit.

artembilan replied:
We take conventions and code style rules mostly from here: https://github.com/spring-projects/spring-framework/wiki/Code-Style#javadoc-formatting

> Use an imperative style (i.e. Return and not Returns) for the first sentence.

Therefore what I want from you is just to change the first word of the method Javadoc from infinitive:

```
Define if a single or a batch of messages are send downstream.
```

tzolov committed:
minor threadfactory nameing fix

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,283 @@
+	/**
+	 * @return Returns current header mapper.
+	 */
+	public HeaderMapper<List<Header<Object>>> getHeaderMapper() {
```
Why do we need this getter?
It does not give any gain from a management perspective and better to design our classes as immutable as possible.

tzolov replied:
because it is handy. check the integration tests

artembilan replied:
Then use `TestUtils.getProperty()`.
Whatever is not intended for end-user must be hidden.
It is really not so handy given an outcome that channel adapter might be mutated.
So, we may got some GH issue with a strange behavior.
And that will be exactly our fault since we have exposed this property.

tzolov replied:
What I meant it that this is usefulul to simplify the debezium-message-producer configuration. Given that later is configured by default with DefaultDebeziumHeaderMapper, e.g. the instance is already there, it is awkward to expect someone to create a new instance from the same class just be able to configure it slitely differntly. 
This was the primary reason I introduced the mirror setAllowedHeaderNames().

artembilan replied:
Well, for consistency with Java Beans spec it is better to keep this as straightforward as possible.
The `setAllowedHeaderNames()` is really nor this component responsibility and even if we inject a new instance of the same class, we still just override its internal value.
The setter way is easy to follow from an XML config and so on.
It is really better to give only one choice to end-users: "want some heads to filter out? - Provide a respective `HeaderMapper`".

When we design this component we need to think about responsibility level and convenience for end-users.
The getter is not so convenient and does not give a clue what we support to do with its result in the target project.
Users of our API really don't care how is that convenient for your tests.

When we introduce Java DSL for this channel adapter, that's where we will expose a convenient `headersToMap()` option.

tzolov replied:
> Users of our API really don't care how is that convenient for your tests.
As if explained, if just someone have read it, It is convenient for the users not for my tests. Anyway feel free to change it as it find it fit

artembilan replied:
Well, this is your PR and it is hard for me to change something especially if you already tie into it in the test.
Moreover this is just a discussion without any pressure.
I just don't feel comfortable to expose a getter for end-user configuration considerations, where existing setter is really just a single place of truth and it should ensure an immutability state of our component after configuration phase.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,283 @@
+	 * @param debeziumBuilder - pre-configured Debezium Engine Builder instance.
+	 */
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
+		Assert.notNull(debeziumBuilder, "The Debezium Engine Builder is null!");
```
Don't yell to end-users.
Just simple ` "'debeziumBuilder' must not be null"` is enough.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,283 @@
+	@Override
+	protected void onInit() {
+		super.onInit();
+		Assert.notNull(this.headerMapper, "Header mapper can not be null!");
```
According to the `setHeaderMapper` logic it cannot be null, so just remove this line from here.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,283 @@
+
+		// If payload is still null ignore the message.
+		if (payload == null) {
+			logger.info("Dropped null payload message");
```
Can we also add here something like `" for:" + changeEvent`?
To have some traceability in the logs to correlate such a not with an event produced by the Debezium.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java:
```diff
@@ -0,0 +1,283 @@
+		}
+
+	}
+}
```
Still missed one new line before the last brace in the class 😉

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/support/DebeziumHeaders.java:
```diff
@@ -0,0 +1,39 @@
+ * Pre-defined header names to be used when retrieving Debezium Change Event headers.
+ *
+ * @author Christian Tzolov
+ * @since 6.2
```
I prefer to separate `@since` from the `@author` for better visibility.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/support/DefaultDebeziumHeaderMapper.java:
```diff
@@ -0,0 +1,95 @@
+		this.allowedHeaderNames = copy;
+	}
+
+	private boolean shouldMapHeader(String headerName, String[] patterns) {
```
Can it be `static`?
Just for consistency of the utility nature of this method.

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/support/DebeziumHeaders.java:
```diff
@@ -0,0 +1,39 @@
+ * limitations under the License.
+ */
+
+package org.springframework.integration.debezium.inbound.support;
```
I think you have missed something.
These headers-related classes must be in the `org.springframework.integration.debezium.support` package - top-level of the module, not a nested one in still that `inbound`.

artembilan commented on spring-integration-debezium/src/test/resources/log4j2-test.xml:
```diff
@@ -0,0 +1,17 @@
+	</Appenders>
+	<Loggers>
+		<Logger name="org.junit" level="warn" />
+		<Logger name="org.testcontainers" level="INFO"/>
```
Sorry, but do we really still need these two categories to manage?

tzolov replied:
Yes i've been using testcontainers log configuration especially when turned in debug mode.

artembilan commented on src/reference/asciidoc/debezium.adoc:
```diff
@@ -0,0 +1,172 @@
+        DebeziumEngine.Builder<ChangeEvent<byte[], byte[]>> debeziumEngineBuilder ) {
+
+        return IntegrationFlow.from(
+                         new DebeziumMessageProducer(debeziumEngineBuilder);)
```
That semicolon must be removed.

artembilan commented on src/reference/asciidoc/whats-new.adoc:
```diff
@@ -17,5 +17,12 @@ In general the project has been moved to the latest dependency versions.
+[[x6.2-debezium]]
+==== Debezium Inbound Channel Adapter
+
+The Debezium Spring Integration Debezium channel adapter is part of the `spring-integration-debezium` module.
```
Something is wrong with this sentence

artembilan commented on src/reference/asciidoc/whats-new.adoc:
```diff
@@ -17,5 +17,12 @@ In general the project has been moved to the latest dependency versions.
+==== Debezium Inbound Channel Adapter
+
+The Debezium Spring Integration Debezium channel adapter is part of the `spring-integration-debezium` module.
+The Debezium Engine based Change Data Capture (CDC) channel adapter, allows capturing database change events, converting them into Messages and streaming those to the outbound channels.
```
May be `allows receiving database`?
Otherwise double "Capture" in this sentence sounds a bit awkward.

tzolov replied:
imo, capturing is better aligned with the CDC terminology

artembilan reviewed (changes requested):
Next round of review.

tzolov committed:
fix support package structure

tzolov committed:
another review round

artembilan commented on spring-integration-debezium/src/main/java/org/springframework/integration/debezium/support/DefaultDebeziumHeaderMapper.java:
```diff
@@ -0,0 +1,95 @@
+	 * @see <a href=
+	 * "https://debezium.io/documentation/reference/2.2/transformations/event-flattening.html#extract-new-record-state-add-headers-prefix">add.headers.prefix</a>
+	 */
+	public void setAllowedHeaderNames(String... headerNames) {
```
Now I paid attention to this method name.
What do we allow or disallow?
There is no security restrictions involved.
Why just not `headersToMap()`?

tzolov committed:
another review resolution round

tzolov committed:
minor spelling fix

artembilan commented:
Merged as https://github.com/spring-projects/spring-integration/commit/8b004e9ec2be349cd112c84ba7075e84f1eaa232 after some code style clean-ups and typos fixing.
